### PR TITLE
Make sure admin events are not referencing sensitive data from their representation (rebased)

### LINKED ISSUE #21562: Make sure admin events are not referencing sensitive data from their representation
### Description

It is a follow-up on https://github.com/keycloak/keycloak/issues/17470.

In addition to removing credentials from user representations when persisting admin events, we should also make sure other representation types are not exposing any sensitive data.

The representation types we should start looking at are:

* `RealmRepresentation`
* `UserRepresentation`
* `ClientRepresentation`
* `IdentityProviderRepresentation`
* `ComponentRepresentation`

Ideally, we should enhance how we are using `StripSecretsUtils` to encapsulate how sensitive data are removed from representations for the different representation types. Therefore, helping to administer how sensitive data, such as secrets, are returned from Admin API and set to event representations.

### Discussion

https://github.com/keycloak/keycloak/issues/17470

### Motivation

Avoid exposing sensitive data when returning representations from Admin API or when creating/persisting events.

### Details

_No response_

### PR DESCRIPTION
**Pull request by ioemat**
closes #21562

### CODE REVIEW

**Comment by ioemat** [comment]
fixes #21562 21562

File: server-spi-private/src/main/java/org/keycloak/models/utils/StripSecretsUtils.java [hunk 1]
```diff
@@ -42,6 +40,35 @@ public class StripSecretsUtils {
...
+        if (formatter == null) {
+            return representation;
+        }
+
+        formatter.accept(session, representation);
+
+        return representation;
+    }
+
     private static String maskNonVaultValue(String value) {
```
Comment (ioemat) [inline server-spi-private/src/main/java/org/keycloak/models/utils/StripSecretsUtils.java:72]: Not sure why the special handling for this is here, is this sth. you use internally at redhat?

**Comment by ioemat** [comment]
Looks like one more rebase is required. I'll do this after receiving feedback

**Comment by jonkoops** [comment]
Hi @ioemat, please [sign off](https://github.com/keycloak/keycloak/pull/26505/checks?check_run_id=20862228196) on your commits. You should also make sure to [link your pull request](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) to the issue, and give it a descriptive title rather than referring to the issue there.

**Comment by ioemat** [comment]
hmm, build failed with gateway timeout, I think it should be restarted

File: server-spi-private/src/main/java/org/keycloak/models/utils/StripSecretsUtils.java (hunk 1 above)
Comment (pedroigor) [reply server-spi-private/src/main/java/org/keycloak/models/utils/StripSecretsUtils.java:72]: Not sure either.

**Review by pedroigor — COMMENTED** [review]
@ioemat Re-running tests but they are probably related?

**Comment by ioemat** [comment]
> @ioemat Re-running tests but they are probably related?

Yes, after rebasing there are some real test errors, I'm on it

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/exportimport/ExportImportTest.java [hunk 2]
```diff
@@ -452,13 +454,6 @@ private void testRealmExportImport() throws LifecycleException {
         // Ensure data are imported back, but just for "test" realm
         Assert.assertNames(adminClient.realms().findAll(), "master", "test", TEST_REALM);
 
-        assertAuthenticated("test", "test-user@localhost", "password");
```
Comment (ioemat) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/exportimport/ExportImportTest.java:455]: @pedroigor I think we need a discussion on this.

When secrets are stripped when exporting e.g. e realm it becomes impossible to migrate a realm by export/import. Not sure whether this is intended.

Comment (ioemat) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/exportimport/ExportImportTest.java:455]: Maybe we should split the PR into use cases where we are sure that secrets should be hidden (e.g. Admin Events) and use cases we need to discuss (e.g. Realm export)

**Comment by ioemat** [comment]
Rebased the PR and reduced it to what the originally topic of 21562 was: Don't expose credentials in events.

Additionally it changes the GET realm resource to masked secrets as well, IMHO this makes sense. What do you think @pedroigor?

**Comment by ioemat** [comment]
> Rebased the PR and reduced it to what the originally topic of 21562 was: Don't expose credentials in events.
> 
> Additionally it changes the GET realm resource to masked secrets as well, IMHO this makes sense. What do you think @pedroigor?

Any thoughts on this @pedroigor?

**Review by pedroigor — APPROVED** [review]
@ioemat Sorry for the delay. LGTM.

Having the logic in a single place as we discussed helps a lot.

**Pull request merged**